src/Time/TimeSteppers/DormandPrince5.cpp

```diff
@@ -0,0 +1,106 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Time/TimeSteppers/DormandPrince5.hpp"
+
+#include <cmath>
+#include <limits>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Time/TimeStepId.hpp"
+
+namespace TimeSteppers {
+
+uint64_t DormandPrince5::number_of_substeps() const noexcept { return 6; }
+
+size_t DormandPrince5::number_of_past_steps() const noexcept { return 0; }
+
+// The growth function for DP5 is
+//
+//   g = z^6 / 600 + \sum_{n=0}^5 mu^n / n!,
+//
+// where mu = lambda * dt. The equation dy/dt = -lambda * y evolves
+// stably if |g| < 1. For lambda=-2, chosen so the stable_step() for
+// RK1 (i.e. forward Euler) would be 1, DP5 has a stable step
+// determined by inserting mu->-2 dt into the above equation. Finding the
+// solutions with a numerical root find yields a stable step of about 1.653.
+double DormandPrince5::stable_step() const noexcept {
+  return 1.6532839463174733;
+}
+
+TimeStepId DormandPrince5::next_time_id(const TimeStepId& current_id,
+                                        const TimeDelta& time_step) const
+    noexcept {
+  switch (current_id.substep()) {
```

Comment:
I'm not sure how possible this is and how @wthrowe feels about it. It looks like a lot of these switch statements are the same in the sense that they depend on `substep` and `substep - 1`. Would it be possible to write this as something like:
```cpp
if (current_id.substep() == 0) {
  ...
} else if (current_id.substep() < 6) {
  ...
} else {
  ERROR
}
```
That could same something like 40 lines of code here